first group project upload

adriverhagen opened the pull request:
Please ignor project 2 & 5, since i have alternative solutions for them.

adriverhagen committed:
first group project upload

santiagobasulto commented:
This PR will be reviewed by: @iamnotmichael

iamnotmichael commented on group_project/assignment_1.py:
```diff
@@ -1,6 +1,7 @@
+    if len(a_string) %2 == 0:
+        return a_string[:len(a_string)/2]
+    return a_string[:(len(a_string)+1)/2]
 
```
Well done

iamnotmichael commented on group_project/assignment_3.py:
```diff
@@ -1,7 +1,10 @@
+        a_string = a_string.replace(word_to_replace, new_word)
+        return a_string
+    return a_string
+    
```
Good job. You did not put a redundant elif like I did. If you haven't, check out Santiagos solution to this in his email. This can be done without an if statement.

iamnotmichael commented on group_project/assignment_4.py:
```diff
@@ -1,5 +1,7 @@
+    if a_word in a_string:
+        return a_string.count(a_word)
+    return 0
 
```
You do not need the if statement. You can simply return a_string.count(a_word).

adriverhagen replied:
But what if the a_word is not in a_string ?

iamnotmichael replied:
It returns a zero automatically if the word is not in it if you do not use an if statement. If you do the If statement route, it will return None so you do have to specify return 0.

def how_many_times(a_string, a_word):
    return a_string.count(a_word)

adriverhagen replied:
Ah cool! thanks for explaining that.

iamnotmichael commented on group_project/assignment_6.py:
```diff
@@ -1,15 +1,32 @@
+    )
+    
+    return board
 
```
Only issues I see is using the Global variable of BOARD_TEMPLATE instead of building your board in your function. Either way, If you go that route, you don’t need to build your board with r1_1, r1_2, etc. You could simple put:

def format_tic_tac_toe_board(first_row, second_row, third_row):

    board = BOARD_TEMPLATE.format(
        first_row[0], first_row[1], first_row[2],
        second_row[0], second_row[1], second_row[2],
        third_row[0], third_row[1], third_row[2],
    )

adriverhagen committed:
updated exercise 2 & 5

	modified:   group_project/assignment_2.py
	modified:   group_project/assignment_5.py

adriverhagen commented:
updated assignment 2 & 5

adriverhagen committed:
updated
